### DISCUSSION TITLE
CacheCleared event

### DISCUSSION THREAD

**Discussion by ryanmitchell** [laravel/framework#50667] [Ideas]
Would you be open to adding (or reviewing a PR for) a CacheCleared event? 

We have a use case where we're keeping record of cache entries in a database store, so when the cache is cleared it gets out of sync. It would be useful to be able to listen for that happening and re-sync / re-populate our store at that point.

**Comment by henzeb** [comment]
do you need the cache entries stored in the database?

**Reply by ryanmitchell** [reply]
No, its just the keys - its part of a larger caching system where we create dependencies between certain cache keys, so if a linked key is cleared then any dependent keys are also cleared.
